bin/release.sh

```diff
@@ -2,23 +2,42 @@
 
 set -e
 
+# Make sure the release tag is provided.
 if (( "$#" != 1 ))
 then
-    echo "Tag has to be provided"
+    echo "Tag has to be provided."
 
     exit 1
 fi
 
-CURRENT_BRANCH="6.x"
+# Make sure the working directory is clear.
+if [[ ! -z "$(git status --porcelain)" ]]
+then
+    echo "Your working directory is dirty. Did you forget to commit your changes?"
+
+    exit 1
+fi
+
+RELEASE_BRANCH="6.x"
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
 VERSION=$1
 
+# Make sure current branch and release branch match.
+if (( $RELEASE_BRANCH != $CURRENT_BRANCH ))
+then
+    echo "Release branch ($RELEASE_BRANCH) does not match the current active branch ($CURRENT_BRANCH)."
+
+    exit 1
+fi
+
 # Always prepend with "v"
 if [[ $VERSION != v*  ]]
 then
     VERSION="v$VERSION"
 fi
 
 # Tag Framework
+git pull
```

Comment:
If between running this script and then running `git pull` something changes (and causes issue) should we stop execution?